### DISCUSSION TITLE
MutualInformationMetric random sampling

### DISCUSSION THREAD

**Discussion by rhacking** [dipy/dipy#1736] [Registration] [answered]
As far as I know, there is currently no way to use `MutualInformationMetric` with random sampling points. Elastix seems to use this as their default (http://elastix.bigr.nl/wiki/images/c/c5/Parameters_Affine.txt), whereas DIPY currently only seems to support full and grid sampling. It would be nice if such a random sampling (every iteration) feature could be implemented into DIPY.

**Comment by Garyfallidis** [comment, marked as answer]
Hi @rhacking, thank you for pointing this out. Elastix and Ants are larger registration projects and they do have more options. However, we are always welcoming new contributions. I hope you can add that option if it is useful to your projects. Maybe useful to others too. We would like some help with adding more metrics too. MSD and CC for affine are currently missing. Let us know if you or other people you may know want to contribute and thanks again for pointing this out.
